lighthouse-core/gather/gatherers/iframe-elements.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const pageFunctions = require('../../lib/page-functions.js');
+
+/* eslint-env browser, node */
+
+/**
+ * @return {LH.Artifacts['IFrameElements']}
+ */
+/* istanbul ignore next */
+function collectIFrameElements() {
+  // @ts-ignore - put into scope via stringification
+  const iFrameElements = getElementsInDocument('iframe'); // eslint-disable-line no-undef
+  return iFrameElements.map(/** @param {HTMLIFrameElement} node */ (node) => {
+    // @ts-ignore - put into scope via stringification
+    const clientRect = node.getBoundingClientRect().toJSON(); // eslint-disable-line no-undef
+    // Marking 1x1 as non-visible to ignore tracking pixels.
+    const isVisible = (clientRect.width > 1 && clientRect.height > 1);
+    return {
+      id: node.id,
+      src: node.src,
+      clientRect,
+      isVisible,
```

Comment:
should this be renamed? This would catch `display: none` (maybe should mention this in the comment over `isVisible` declaration), but not `opacity` or if it's occluded by other elements, etc. What's the use of the property, exactly? (maybe "is this a frame we might care about"?)

Reply:
Following Paul's pixelArea idea, so no longer applicable.